apps/explorer/src/ui/Button.tsx

```diff
@@ -35,10 +36,20 @@ const buttonStyles = cva(
 
 export interface ButtonProps
     extends VariantProps<typeof buttonStyles>,
-        ButtonOrLinkProps {}
+        ButtonOrLinkProps {
+    loading?: boolean;
+}
 
-export function Button({ variant, size, ...props }: ButtonProps) {
+export function Button({
+    variant,
+    size,
+    loading,
+    children,
+    ...props
+}: ButtonProps) {
     return (
-        <ButtonOrLink className={buttonStyles({ variant, size })} {...props} />
+        <ButtonOrLink className={buttonStyles({ variant, size })} {...props}>
```

Comment:
Maybe disable the button when it's in `loading` state? Could help prevent double submission.